src/obj-ignore.c

```diff
@@ -175,16 +175,16 @@ static void rune_add_autoinscription(struct object *obj, int i)
 
 	/* No autoinscription, or already there, don't bother */
 	if (!rune_note(i)) return;
-	if (obj->note && strstr(quark_str(obj->note), quark_str(rune_note(i))))
+	if (obj->note && strstr(obj->note, rune_note(i)))
 		return;
 
 	/* Extend any current note */
 	if (obj->note)
-		my_strcpy(current_note, quark_str(obj->note), sizeof(current_note));
-	my_strcat(current_note, quark_str(rune_note(i)), sizeof(current_note));
+		my_strcpy(current_note, obj->note, sizeof(current_note));
+	my_strcat(current_note, rune_note(i), sizeof(current_note));
 
 	/* Add the inscription */
-	obj->note = quark_add(current_note);
+	obj->note = string_make(current_note);
```

Comment:
It look like a `string_free(obj->note);` is needed immediately before this to avoid a memory leak.